subsystems/controllers/fade_in_controller.py

```diff
@@ -0,0 +1,25 @@
+from wpilib._wpilib import Timer
+from utils import math_util
+
+class FadeInController:
+    def __init__(self, wanted_color, duration):
+        self.timer = Timer()
+        self.h = wanted_color[0]
+        self.s = wanted_color[1]
+        self.v = wanted_color[2]
+
+        self.duration = duration
+        self.timer.reset()
+
+    def update(self, led_buffer):
+        n = 1 - ((self.timer.get() % self.duration) / self.duration)
+
+        for d in led_buffer:
+            d.setHSV(self.h, self.s, math_util.int(self.v * n))
+
+        return led_buffer
+
+    def isFinished(self):
+        if self.duration <= self.timer.get():
+            return True
+        return False
```

Comment:
Really minor change, there should be a newline should be at the end of each file. PyCharm has a way to reformat files to fit with python's standards. To do this type `shift` `shift` to open up a menu and type `reformat code` and it should be the first option there.

Reply:
ok Thanks!